src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -166,16 +164,16 @@ public class NetworkPeerConnection
         /// Initializes an instance of the object.
         /// </summary>
         /// <param name="peer">Network peer the node is connected to.</param>
-        /// <param name="socket">Connected network socket to the peer.</param>
+        /// <param name="client">Connected network client to the peer.</param>
         /// <param name="dateTimeProvider">Provider of time functions.</param>
         /// <param name="loggerFactory">Factory for creating loggers.</param>
-        public NetworkPeerConnection(NetworkPeer peer, Socket socket, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
+        public NetworkPeerConnection(NetworkPeer peer, NetworkPeerClient client, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName, $"[{peer.PeerAddress.Endpoint}] ");
```

Comment:
Do we want to guard against peer or peer.PeerAddress being null?

Reply:
hmm, i could maybe add guards over the whole factory interface, good idea, tx
i'll add it as a part of PR part 14 if you don't insist on having it here

Reply:
Wow part 14... this is epic :grin: Sounds good!

Reply:
How many seasons more are you planing 😆

Reply:
There is still a lot to do, consider we don't still don't have any interface on what was node ...